File: code_samples/haskell/error_handling_logging.hs
-- Error Handling: Logging
-- AI/ML Training Sample

module Logging where

data Logging = Logging {
    getData :: String
} deriving (Show, Eq)

process :: Logging -> String -> Logging
process obj input = obj { getData = input }

validate :: Logging -> Bool
validate obj = not (null (getData obj))

main :: IO ()
main = do
    let instance = Logging { getData = "" }
    let updated = process instance "example"
    putStrLn $ "Data: " ++ getData updated
    putStrLn $ "Valid: " ++ show (validate updated)

File: code_samples/lua/error_handling_logging.lua
-- Error Handling: Logging
-- AI/ML Training Sample

Logging = {}
Logging.__ index = Logging

function Logging:new()
    local obj = {
        data = ""
    }
    setmetatable(obj, self)
    return obj
end

function Logging:process(input)
    self.data = input
end

function Logging:getData()
    return self.data
end

function Logging:validate()
    return #self.data > 0
end

-- Example usage
local instance = Logging:new()
instance:process("example")
print("Data: " .. instance:getData())
print("Valid: " .. tostring(instance:validate()))
